### DISCUSSION TITLE
FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:SetInputs:TwrInfl:getLocalTowerProps:Tower strike.

### DISCUSSION THREAD

**Discussion by tatesnow** [OpenFAST/openfast#960] [Q&A] [answered]
When I simulate with 25m/s turbulent wind, the following error occurs, can you give me some suggestions?

> OpenFAST-v3.0.0
Compile Info:
 - Compiler: Intel(R) Fortran Compiler 1900
 - Architecture: 64 bit
 - Precision: single
 - OpenMP: No
 - Date: Jun 23 2021
 - Time: 21:25:05
Execution Info:
 - Date: 01/06/2022
 - Time: 17:13:27+0800

OpenFAST input file heading:
    FAST          Certification - Test #18: NREL 5.0 MW Baseline Wind Turbine (Onshore)

Running ElastoDyn.
Nodal outputs section of ElastoDyn input file not found or improperly formatted.
Running AeroDyn.
AD15 Nodal Outputs: Nodal output section of AeroDyn input file not found or improperly formatted.
Skipping nodal outputs.
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 1, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 2, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 3, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 4, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 1, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 2, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 3, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 4, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 1, blade 3)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 2, blade 3)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 3, blade 3)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 4, blade 3)
Running InflowWind.

   Reading a 39x34 grid (190 m wide, 7.5 m to 172.5 m above ground) with a characteristic wind
   speed of 25 m/s. This full-field file was generated by TurbSim (v2.00.07a-bjj, 14-Jun-2016) on
   04-Jan-2022 at 20:24:06.

   Processed 12420 time steps of 20-Hz full-field data (period of 621 seconds).
Running ServoDyn.
Running ServoDyn Interface for Bladed Controllers (using Intel Visual Fortran for Windows, ).
Using legacy Bladed DLL interface.
 Time: 0 of 60 seconds.

FAST_Solution0:CalcOutputs_And_SolveForInputs:SolveOption2:SrvD_CalcOutput:DLL_controller_call:Run
ning with torque and pitch control of the NREL offshore 5MW baseline wind turbine from DISCON.dll
as written by J. Jonkman of NREL/NWTC for use in the IEA Annex XXIII OC3 studies.

FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption2:RotCalcOutput:BEMT_CalcOutput(node 5,
blade 1):UA_CalcOutput:UA_BlendSteady:Temporarily turning off UA due to high angle of attack or
low relative velocity. This warning will not be repeated though the condition may persist.

FAST_Solution:FAST_AdvanceStates:ED_ABM4:ED_CalcContStateDeriv:SetCoordSy:Small angle assumption
violated in SUBROUTINE SmllRotTrans() due to a large blade deflection (ElastoDyn SetCoordSy). The
solution may be inaccurate. Simulation continuing, but future warnings from SmllRotTrans() will
be suppressed.
 Additional debugging message from SUBROUTINE SmllRotTrans(): 0.5125 s

FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption2:RotCalcOutput:BEMT_CalcOutput(node 19,
blade 2):UA_CalcOutput:Mach number exceeds 0.3. Theory is invalid. This warning will not be
repeated though the condition may persist.

FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:SetInputs:TwrInfl:getLocalTowerProps:Tower
strike.

 OpenFAST encountered an error at simulation time 0.7625 of 60 seconds.
 Simulation error level: FATAL ERROR

 Aborting OpenFAST.

**Comment by jjonkman** [comment, marked as answer]
Dear @tatesnow,

Your OpenFAST model appears to be numerically unstable.  It looks like you are running the NREL 5-MW land-based wind turbine example.  Can you clarify what you changed in this model relative to the model available from the r-test: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_Land_DLL_WTurb?

Best regards,

**Reply by tatesnow** [reply]
Dear @jjonkman,
I only modified the turbulent wind because I used turbSim to generate a 25m/s turbulent wind.
Best regards,

**Reply by tatesnow** [reply]
![image](https://user-images.githubusercontent.com/42032942/148478818-4dcc9af1-c441-4406-b4bf-8c10e480c490.png)

**Reply by tatesnow** [reply]
![image](https://user-images.githubusercontent.com/42032942/148503234-aa5e71e9-9ef0-43ac-9faf-86d7b289f0e0.png)
![image](https://user-images.githubusercontent.com/42032942/148503345-15031441-1c72-4cb7-a56d-de37a0af434a.png)
![image](https://user-images.githubusercontent.com/42032942/148503363-d65e5851-be4f-463a-91e6-abdf0ed0e611.png)

I guess it might be a problem with the shutdown control strategy of the wind turbine?

**Reply by tatesnow** [reply]
Is there a controller that considers the shutdown strategy?

**Reply by jjonkman** [reply]
Dear @tatesnow,

It looks like you changed the mean wind speed of the simulation without setting appropriate values for the initial rotor speed and initial blade-pitch angles.  It is generally recommended to set the initial rotor speed and initial blade-pitch angles to their respective mean values associated with the mean wind speed being simulated.  Otherwise, the speed and pitch may be outside a region the controller was designed for and result in unstable control response and never recover (worst-case scenario), or a start-up transient that is unrealistic and lasts a long time.  In your case, the controller response never recovers.  At 25 m/s for the NREL 5-MW baseline wind turbine, the rotor speed should be initialized to 12.1 rpm and the blade-pitch angles should be initialized to about 23.5 deg in ElastoDyn.  At other mean wind speeds, the recommend initial conditions can be found from Figure 9-1 of the NREL 5-MW baseline wind turbine specifications report: https://www.nrel.gov/docs/fy09osti/38060.pdf.

Best regards,

**Reply by tatesnow** [reply]
Dear @jjonkman,
Get it! Thank you very much!
Best regards,

**Reply by tatesnow** [reply]
Dear @jjonkman ,
According to your suggestions, I modified the initial speed and pitch angle of the input file, and then it works smoothly.
But there are still some warming information, although it does not affect the result, but I still want to know the reason
------
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 1, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 2, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 3, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 4, blade 1)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 1, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 2, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 3, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 4, blade 2)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 1, blade 3)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 2, blade 3)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 3, blade 3)
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0. (node 4, blade 3)

**Reply by jjonkman** [reply]
Dear @tatesnow,

Regarding the "C_nalpha is 0" warnings, you can ignore these, as discussed in a previous OpenFAST issue: https://github.com/OpenFAST/openfast/issues/268.

Best regards,

**Reply by lin990316** [reply]
Dear @jjonkman 
I noticed in your reply about the initial rotor speed and initial blade angle settings for different wind speeds, which I seem to have missed in my previous calculations, but I don't seem to have found any detailed information about this for the IEA 15MW turbine that I am studying. Can you tell me if the table in Fig. 9.1 that you mentioned applies to other turbines, and if this figure doesn't apply to other turbines, where can I find it? Where can I find pictures of the 15MW wind turbine?
Best regards,

**Reply by jjonkman** [reply]
Dear @lin990316,

It looks like you asked a similar question in the following OpenFAST issue: https://github.com/OpenFAST/openfast/issues/2047; I've responded there.

Best regards,

**Comment by Crystal0912** [comment]
I have a similar question with you .When I simulate with 12m/s turbulent wind, the following error occurs, can you give me some suggestions?
**************************************************************************************************
 FAST (v8.16.00a-bjj, 27-Jul-2016)
 Copyright (C) 2016 National Renewable Energy Laboratory
 This program comes with ABSOLUTELY NO WARRANTY. See the "license.txt" file distributed with this
 software for details.
 **************************************************************************************************
  Running FAST (v8.16.00a-bjj, 27-Jul-2016), compiled as a DLL S-Function for Simulink as a 64-bit
  application using single precision
  linked with NWTC Subroutine Library (v2.09.00, 23-Jul-2016)
  Heading of the FAST input file:
    FAST Certification Test #19: NREL 5.0 MW Baseline Wind Turbine with OC3 Monopile RF
    Configuration, for use in offshore analysis
  Running ElastoDyn (v1.04.00a-bjj, 26-Jul-2016).
  Running AeroDyn (v15.03.00, 27-Jul-2016).
  Running AirfoilInfo (v1.01.00a-bjj, 5-Apr-2016).
  Running BEM (v1.02.00, 29-Jun-2016).
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 1, Blade = 1
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 2, Blade = 1
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 3, Blade = 1
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 4, Blade = 1
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 1, Blade = 2
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 2, Blade = 2
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 3, Blade = 2
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 4, Blade = 2
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 1, Blade = 3
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 2, Blade = 3
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 3, Blade = 3
 Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 4, Blade = 3
  Running InflowWind (v3.03.00, 26-Jul-2016).
  Opening InflowWind input file:  .\5MW_Baseline/NRELOffshrBsline5MW_InflowWind_12mps.dat
    Reading a 15x15 grid (145 m wide, 17.5 m to 162.5 m above ground) with a characteristic wind
    speed of 18 m/s. This full-field file was generated by TurbSim (v1.06.00, 21-Sep-2012) on
    23-Nov-2021 at 20:32:24.
    Processed 12162 time steps of 20-Hz full-field data (608.05 seconds).
  Running ServoDyn (v1.06.00a-bjj, 26-Jul-2016).
  Running ServoDyn Interface for Bladed Controllers (using Intel Visual Fortran for
  Windows/Matlab, 14-Oct-2015).
  Running HydroDyn (v2.05.01, 27-Jul-2016).
  Generating incident wave kinematics and current time history.
  Calculating second order difference frequency wave kinematics.
  Calculating second order sum frequency wave kinematics.
  Running SubDyn (v1.03.00, 11-Apr-2016).
    Calculating Internal Modal Eigenvectors

FAST_InitializeAll:SrvD_Init:ValidatePrimaryData:Yaw angle and rate are not commanded from Simulink model.
ValidatePrimaryData:Pitch angles are not commanded from Simulink model.
ValidatePrimaryData:Generator torque and power are not commanded from Simulink model.
ValidatePrimaryData:HSS brake is not commanded from Simulink model.
  Timestep: 0 of 100 seconds.
  FAST completed.
An error occurred while running the simulation and the simulation was terminated
Caused by:
Error reported by S-function 'FAST_SFunc' in '[Program/FAST Nonlinear Wind Turbine/S-Function](matlab:open_and_hilite_hyperlink ('Program/FAST Nonlinear Wind Turbine/S-Function','error'))':
FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:BEMT_UpdateStates(node 10, blade 1):BEMT_UnCoupledSolve:DeterminePhiBounds:There is no valid value of phi for these operating conditions!  Vx = -5.5312, Vy = 45.188, rlocal = 32.232, theta = 0.18825

**Comment by Crystal0912** [comment]
Especially for this :---Caused by:
Error reported by S-function 'FAST_SFunc' in 'Program/FAST Nonlinear Wind Turbine/S-Function':
FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:BEMT_UpdateStates(node 10, blade 1):BEMT_UnCoupledSolve:DeterminePhiBounds:There is no valid value of phi for these operating conditions! Vx = -5.5312, Vy = 45.188, rlocal = 32.232, theta = 0.18825

**Reply by jjonkman** [reply]
Dear @Crystal0912,

You haven't said what you changed in Test 19 of the FAST v8.16 CertTest, but an error regarding "no valid value of phi" is likely indicative of a problem with your simulation set up that prevents blade-element/momentum (BEM) theory from returning a valid solution.  Upgrading your model from FAST v8.16 to OpenFAST is one solution because the BEM algorithm of AeroDyn has been made much more robust in OpenFAST such that this error does not exist anymore.

FYI: The equivalent of Test 19 of the FAST v8.16 CertTest is the following r-test in OpenFAST: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_OC3Mnpl_DLL_WTurb_WavesIrr.

Best regards,

**Reply by Crystal0912** [reply]
Dear @jjonkman,
Thank you for your advice!
Best regards,

**Reply by Crystal0912** [reply]
Dear @jjonkman,
When modifying the parameters "BMassDen, FlpStff, EdgStff" in the “Blade.dat” file to simulate the blade structure damage, what should be paid attention to and why are errors always reported as fllows：
"ReadBladeFile:Invalid numerical input for file ".\5MW_Baseline/NRELOffshrBsline5MW_Blade_Fault2.dat" occurred while trying to read BldFl1Sh(1)."

**Reply by jjonkman** [reply]
Dear @Crystal0912,

Did you add or remove lines from the DISTRIBUTED BLADE PROPERTIES table of the ElastoDyn blade input file without changing `NBlInpSt` accordingly?  Regardless, as with any input file processing error, I would suggest enabling the `Echo` option in ElastoDyn to debug.

Best regards,

**Reply by Crystal0912** [reply]
Dear @jjonkman,

Thank you for your reply. I didn't add or delete lines. I multiply the corresponding number of "BMassDen, FlpStff, EdgStff" in line 11-20 by 0.8, and the operation is successful. But multiply it by other lines in the same way, and an error will be reported. So I want to know if there is any problem when grouping 49 lines of numbers? How many categories should it be divided into? I read that your article is divided into 9 or 10 nodes. How do these 49 lines correspond to nodes?
<img width="388" alt="c543d429d8a1bc380c1b5ce4c5f7c18" src="https://user-images.githubusercontent.com/124583351/223306290-88142e32-9f1a-4e1b-93a6-99cef31e54b3.png">

**Reply by Crystal0912** [reply]
Dear @jjonkman,

Can you provide some suggestions  for simulating blade structural damage？


Best regards,

**Reply by jjonkman** [reply]
Dear @Crystal0912,

I'm not really understanding what changes you are making to the ElastoDyn blade input file that result in errors.  Please clarify your issue or use the `Echo` option to debug problems within input file formatting.

When AeroDyn v15 is enabled, which it looks like you are using, the Figure you show is not valid.  When AeroDyn v15 is enabled, the aerodynamic blade nodes in AeroDyn v15 are distinct from the structural blade nodes in ElastoDyn.  When AeroDyn v15 is enabled, the aerodynamic blade nodes are defined within the AeroDyn v15 blade input file and the structural blade nodes are defined by splitting the blade into `BldNodes` equally-spaced elements, with the nodes located at the centers of those elements.  The blade structural data is specified in the ElastoDyn blade input file are interpolated to these structural analysis nodes; you can see the locations of the blade structural nodes and the resulting interpolated data in the ElastoDyn summary file.

Regarding the modeling of blade structural damage, this topic has been discussed on our forum, e.g., see:
https://forums.nrel.gov/t/making-fault/1260
https://forums.nrel.gov/t/blaide-failure-in-fast/1916

Best regards,

**Reply by Crystal0912** [reply]
Dear @jjonkman,

Thanks for your kind help and timely responses all the time! I will learn from the link you gave.

Best regards,

**Reply by Crystal0912** [reply]
Dear @jjonkman,

Sorry to bother you so frequently. I really use AeroDyn v15. Through learning, I find that if I want to simulate blade structural damage（Blade fault ）, I should change the  factors “AdjBlMs, AdjFlSt，AdjEdSt and BMassDen ，FlpStff，EdgStff ”. Am I right？If I understand correctly, how should I revise these parameters? What is their relationship? How to avoid that this error "Error occurred while trying to read BldFl1Sh (i)(which is in 'BLADE MODE SHAPES')" does not occur? In other words, where can I learn about the relationship between these parameters in the "Blade. dat" file? By the way， I use FASTv8.

**Reply by jjonkman** [reply]
Dear @Crystal0912,

Again, I can't provide more guidance on your input file processing error without more clarification from you, but you should be able to use the `Echo` option to debug problems within input file formatting.

I would not expect structural damage to effect the blade mass.  It could effect the blade stiffness, but I'm not an expert on what such relationships would be.  Perhaps someone else can comment.

Best regards,

**Reply by Crystal0912** [reply]
Dear @jjonkman,

Thank you for your help. I ran successfully by modifying the parameter "FlpStff and EdgStff". It is really irrelevant to the "BMassDen" parameter, as you said.

Best regards,

**Reply by Jafariasl** [reply]
Hi @Crystal0912 and @jjonkman 
I faced this problem. Can you share the modified values of FlpStff and EdgStff?
Best

**Comment by akmalngb** [comment]
Dear @jjonkman I have a similar question. I simulate an 18.9m/s turbulent wind, the following error occurs, can you give me some suggestions?
**************************************************************************************************
FAST (v8.16.00a-bjj, 27-Jul-2016)

Copyright (C) 2016 National Renewable Energy Laboratory

This program comes with ABSOLUTELY NO WARRANTY. See the "license.txt" file distributed with this
software for details.
**************************************************************************************************

 Running FAST (v8.16.00a-bjj, 27-Jul-2016), compiled as a 64-bit application using single
 precision
 linked with NWTC Subroutine Library (v2.09.00, 23-Jul-2016)

 Heading of the FAST input file:
   FAST Certification Test #21: NREL 5.0 MW Baseline Offshore Turbine with OC4 Jacket
   Configuration

 Running ElastoDyn (v1.04.00a-bjj, 26-Jul-2016).

 Running AeroDyn (v15.03.00, 27-Jul-2016).

 Running AirfoilInfo (v1.01.00a-bjj, 5-Apr-2016).

 Running BEM (v1.02.00, 29-Jun-2016).
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 1, Blade = 1
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 2, Blade = 1
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 3, Blade = 1
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 4, Blade = 1
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 1, Blade = 2
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 2, Blade = 2
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 3, Blade = 2
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 4, Blade = 2
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 1, Blade = 3
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 2, Blade = 3
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 3, Blade = 3
Warning: Turning off Unsteady Aerodynamics because C_nalpha is 0.  BladeNode = 4, Blade = 3

 Running InflowWind (v3.03.00, 26-Jul-2016).
 Opening InflowWind input file:  .\NRELOffshrBsline5MW_InflowWind_12mps.dat

   Reading a 20x20 grid (140 m wide, 20 m to 160 m above ground) with a characteristic wind speed
   of 18.91 m/s. This full-field file was generated by TurbSim (v1.06.00, 21-Sep-2012) on
   24-Apr-2025 at 03:25:15.

   Processed 12149 time steps of 20-Hz full-field data (607.4 seconds).

 Running ServoDyn (v1.06.00a-bjj, 26-Jul-2016).

 Running ServoDyn Interface for Bladed Controllers (using Intel Visual Fortran for Windows,
 14-Oct-2015).

 Running HydroDyn (v2.05.01, 27-Jul-2016).
 Generating incident wave kinematics and current time history.

 Running SubDyn (v1.03.00, 11-Apr-2016).
   Calculating Internal Modal Eigenvectors
 Timestep: 0 of 600 seconds.

FAST_Solution0:CalcOutputs_And_SolveForInputs:SolveOption2:SrvD_CalcOutput:Running with torque
and pitch control of the NREL offshore 5MW baseline wind turbine from DISCON.dll as written by J.
Jonkman of NREL/NWTC for use in the IEA Annex XXIII OC3 studies.


FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption2:SrvD_CalcOutput:BladedInterface option
was designed for an explicit-loose coupling scheme. Using last calculated values from DLL on all
subsequent calls until time is advanced. Warning will not be displayed again.


FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption1:ED_SD_HD_BD_Orca_InputOutputSolve:SD_Cal
cOutput:Small angle assumption violated in SUBROUTINE SmllRotTrans() due to a large UL input
angles. The solution may be inaccurate. Simulation continuing, but future warnings from
SmllRotTrans() will be suppressed.
 Additional debugging message from SUBROUTINE SmllRotTrans():
FAST_Solution:CalcOutputs_And_SolveForInputs:SolveOption2:AD_CalcOutput:BEMT_CalcOutput(node 18,
blade 2):Compute_UA_AirfoilCoefs:UA_CalcOutput:Mach number exceeds 1.0. Equations cannot be
evaluated.

 FAST encountered an error at simulation time 0.2 of 600 seconds.
 Simulation error level: FATAL ERROR

 Aborting FAST.
 
 And below are my inputs:
 *For elastodyn, I have already set the initial condition for rotor speed and blade pitch based on the Figure 9-1
![image](https://github.com/user-attachments/assets/7f65d590-76e5-4482-9c32-7f96447460ec)
![image](https://github.com/user-attachments/assets/80cf10fd-88f7-4a71-82b9-2890019f377d)

*For TURBISM, I set the wind speed at 18.91m/s
![image](https://github.com/user-attachments/assets/860a1407-14d6-403e-975d-4ec1f98fbe3d)

Can you help me with the solution for the above problem?

Best regard,

**Reply by jjonkman** [reply]
Dear @akmalngb,

It looks like your FAST model is unstable.  Other than the wind speed, rotor speed, and blade-pitch angles, did you change anything in the NREL 5-MW atop OC4-jacket model provided in the CertTest of FAST v8.16?

Best regards,

**Reply by akmalngb** [reply]
Thank you for your response @jjonkman,

I only change DISCON_win32.dll into DISCON_x64.dll at servodyn. Can this affects the problem occured?

**Reply by jjonkman** [reply]
I guess that depends on which version of the DISCON controller you are using; where did you obtain it from?

**Reply by akmalngb** [reply]
I obtained it from "\fast\CertTest\5MW_Baseline\ServoData\DISCON_x64.dll" provided

**Reply by jjonkman** [reply]
That should be fine.  Does the original model without the wind speed, rotor speed, and blade-pitch angles changes run as expected?

**Reply by akmalngb** [reply]
Since the wind speed should be greater than 0, I used 1 m/s wind speed on TURBISM and set the rotor speed and blade pitch angle to 0. However, I received this error instead:

FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:BEMT_UpdateStates(node 7, blade
3):BEMT_UnCoupledSolve:DeterminePhiBounds:There is no valid value of phi for these operating
conditions!  Vx = 0.1035, Vy = -0.77918, rlocal = 19.938, theta = 0.17734

**Reply by jjonkman** [reply]
The NREL 5-MW baseline wind turbine is not mean to operate at 1 m/s.

The original model of the NREL 5-MW turbine atop the OC4-jacket provided in the OpenFAST r-test used turbulent wind at 12 m/s.  Does this model run for you as expected?

**Reply by akmalngb** [reply]
Ah it works well, I have tried to reanalizing my simulation's input and it turned out that the program had an issue with the time step integration that I used. The simulation that I tried have ran as expected by using 0.001 time step. 

Thank you so much @jjonkman for the solution that you suggested so that the previous problem have been solved.

Best regard,
